File: src/data/structs/service.ts
import { Filter } from './filter';
import { Operator } from '../datatypes/operator';

export interface Service<T, K> {

    operator: Operator;

    add(value: T): Promise<T>;
    get(value: K): Promise<T>;
    remove(value: K): Promise<T>;
    update(value: T): Promise<T>;

    getMany(value: Filter): Promise<T[]>;
    


}

File: Structures/ServicesContracts/Service.cs
using System;
using System.Collections.Generic;
using Comies.Structures.Filters;
using Comies.Structures.Interfaces;

namespace Comies.Structures.ServicesContracts
{
    public interface Service <Structure, FilterType> where Structure : class where FilterType : Filter
    {
        Response<IEnumerable<Structure>> GetAll();
        Response<IEnumerable<Structure>> GetSome(FilterType filter);
        Response<Structure> GetOne(Guid Id);
        Response<Structure> Save(Structure entity);
        Response<Structure> Update(Structure entity);
        Response<Structure> Remove(int Id);
    }
}
